contracts/cw1-subkeys/src/contract.rs

```diff
@@ -88,6 +90,44 @@ where
                     allowance.balance = allowance.balance.sub(amount.clone())?;
                     allowances.save(owner_raw.as_slice(), &allowance)?;
                 }
+                CosmosMsg::Staking(StakingMsg::Delegate {
+                    validator: _,
+                    amount,
+                }) => {
+                    if !allowance.permissions.delegate {
+                        return Err(StdError::generic_err(PermissionErr::Delegate{}));
+                    }
+                    // Decrease allowance
+                    allowance.balance = allowance.balance.sub(amount.clone())?;
```

Comment:
I would keep the staking items independent of balances. You can delegate and undelegate all you want, but not take money from the account. By requiring the balances here, I need to give someone the power to take my money to let them select the validators I invest in. Just the first check is needed (and see above comment about refactor to make it simpler)